consoleme/handlers/v2/notifications.py

```diff
@@ -0,0 +1,159 @@
+    around policy generation from CloudTrail errors.
+    """
+
+    async def get(self):
+        try:
+            # Check notifications for user, return notifications for user
+            notification_response: GetNotificationsForUserResponse = (
+                await get_notifications_for_user(self.user, self.groups)
+            )
+            notifications: List[
+                ConsoleMeUserNotification
+            ] = notification_response.notifications
+            response = WebResponse(
+                status="success",
+                status_code=200,
+                data={
+                    "unreadNotificationCount": notification_response.unread_count,
+                    "notifications": notifications,
+                },
+            )
+            self.write(response.json())
+        except Exception as e:
+            sentry_sdk.capture_exception()
+            self.set_status(500)
+            response = WebResponse(
+                status=Status2.error, status_code=500, errors=[str(e)], data=[]
+            )
+            self.write(response.json())
+            return
+
+    async def post(self):
+        # Create a notification
+        raise NotImplementedError()
+
+    async def put(self):
+        """
+        Allows an "authorized user" (Any user the notification is intended for) to mark the notification as read/unread
+        or hidden/unhidden for themselves or all other notification recipients
+
+        :return:
+        """
+        change = ConsoleMeNotificationUpdateRequest.parse_raw(self.request.body)
+        errors = []
+
+        for untrusted_notification in change.notifications:
+            notification = await fetch_notification(
+                untrusted_notification.predictable_id
+            )
+            if not notification:
+                errors.append("Unable to find matching notification")
+                continue
+            authorized = is_in_group(
+                self.user, self.groups, notification.users_or_groups
+            )
+            if not authorized:
+                errors.append(
+                    f"Unauthorized because user is not associated with notification: {notification.predictable_id}"
+                )
+                continue
+            if (
+                change.action
+                == ConsoleMeNotificationUpdateAction.toggle_read_for_current_user
+            ):
+                if self.user in notification.read_by_users:
+                    # Mark as unread
+                    notification.read_by_users.remove(self.user)
+                else:
+                    # Mark as read
+                    notification.read_by_users.append(self.user)
+            elif (
+                change.action
+                == ConsoleMeNotificationUpdateAction.toggle_read_for_all_users
+            ):
+                if notification.read_by_all:
```

Comment:
This section could be simplified to `notification.read_by_all = !notification.read_by_all`. I prefer that brevity here because there's a lot of other logic going on around it.

Reply:
simplified